aws/resource_aws_secretsmanager_secret.go

```diff
@@ -0,0 +1,316 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsSecretsManagerSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSecretsManagerSecretCreate,
+		Read:   resourceAwsSecretsManagerSecretRead,
+		Update: resourceAwsSecretsManagerSecretUpdate,
+		Delete: resourceAwsSecretsManagerSecretDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"kms_key_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"recovery_window_in_days": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      30,
+				ValidateFunc: validation.IntBetween(7, 30),
+			},
+			"rotation_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"rotation_lambda_arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"rotation_rules": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"automatically_after_days": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+			"tags": {
```

Comment:
https://github.com/terraform-providers/terraform-provider-aws/blob/44af183718e5ddf44de518a04fd858f84cb350df/aws/tags.go#L20:6

should this use the common schema?

Reply:
Sure! Will update. 👍